### PR TITLE
feat(developer): isRTL support for lexical model editor

### LINKED ISSUE #2556: feat(developer): Support isRTL for lexical models
Per #2554, Keyman Developer lexical model editor needs UI to be able to set the lexical model property of `isRTL`. Compiler and parser need to be validated as well.

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #2556.

This adds support for `isRTL`, `insertAfterWord`, and `quotesForKeepSuggestions` punctuation options for lexical models, both in the parsing and in the user interface. A number of options are presented to the user for the quotes and word breaking characters.

The parser continues to be a regex-based typescriptish parser. Expect it to be gross. It's too big a job to do properly right now, so it is somewhat limited and will bail if it finds structures it can't understand.

I would hope to use something a bit smarter in the future to interface with the editor. Probably when we are actually fully web-based and can depend on a Typescript compiler directly.

Please don't get mad at me! 😟😟😟

### CODE REVIEW

File: windows/src/developer/TIKE/child/Keyman.Developer.UI.UfrmModelEditor.pas [hunk 1]
```diff
@@ -161,19 +179,107 @@ function FormatToIndex(format: TLexicalModelFormat): Integer;
...
+    class procedure SetValue(const opts: array of TComboStringOption;
+      const text: string; combo: TComboBox); static;
+  end;
+
+const
+  CInsertAfterWordOptions: array[0..3] of TComboStringOption = (
+    (value: ' '; name: '(space U+0020)'),
+    (value: ''; name: '(no word break)'),
+    (value: Char($0F0B); name: '(Tibetan tsheg U+0F0B)'),
+    (value: Char($1361); name: '(Ethiopian wordspace U+1361)')
```
Comment (darcywong00) [inline windows/src/developer/TIKE/child/Keyman.Developer.UI.UfrmModelEditor.pas:213]: Would some languages want to use a zero-width space?

**Comment by darcywong00** [comment]
Does this also update .model_info?

File: windows/src/developer/TIKE/child/Keyman.Developer.UI.UfrmModelEditor.pas [hunk 2]
```diff
@@ -161,19 +179,107 @@ function FormatToIndex(format: TLexicalModelFormat): Integer;
...
+    (value: '»'),
+    (value: '"'),
+    (value: ''''),
+    (value: '‹'),
+    (value: '‘'),
+    (value: '‚'),
+    (value: '›'),
+    (value: '「'),
+    (value: '『')
+  );
```
Comment (jahorton) [inline windows/src/developer/TIKE/child/Keyman.Developer.UI.UfrmModelEditor.pas:229]: Noting this and how it's used elsewhere... am I correct in inferring that Developer's editor only supports these options for a model's "open quote" parameter?

**Review by jahorton — COMMENTED** [review]
> The parser continues to be a regex-based typescriptish parser. Expect it to be gross. It's too big a job to do properly right now, so it is somewhat limited and will bail if it finds structures it can't understand.

Ah, I assume that's the role of `LexicalModelParser.pas`?  If so, that'd explain a few of my lingering questions here.

File: windows/src/developer/TIKE/child/Keyman.Developer.UI.UfrmModelEditor.pas (hunk 2 above)
Comment (mcdurdin) [reply windows/src/developer/TIKE/child/Keyman.Developer.UI.UfrmModelEditor.pas:229]: Those are default options but you can type your own if you prefer.

**Comment by mcdurdin** [comment]
> Does this also update .model_info?

No, that's entirely separate -- can you open an issue on that?

File: windows/src/developer/TIKE/child/Keyman.Developer.UI.UfrmModelEditor.pas (hunk 1 above)
Comment (mcdurdin) [reply windows/src/developer/TIKE/child/Keyman.Developer.UI.UfrmModelEditor.pas:213]: Yes, probably, good idea.

File: windows/src/developer/TIKE/child/Keyman.Developer.UI.UfrmModelEditor.dfm [hunk 3]
```diff
@@ -185,8 +189,40 @@ inherited frmModelEditor: TfrmModelEditor
...
+            Caption = 'open'
+            FocusControl = cbOpenQuote
+          end
+          object lblCloseQuote: TLabel
+            Left = 220
+            Top = 110
+            Width = 26
+            Height = 13
+            Caption = 'close'
+            FocusControl = cbOpenQuote
```
Comment (darcywong00) [inline windows/src/developer/TIKE/child/Keyman.Developer.UI.UfrmModelEditor.dfm:222]: Would this be cbCloseQuote?

File: windows/src/developer/TIKE/child/Keyman.Developer.UI.UfrmModelEditor.dfm [hunk 4]
```diff
@@ -211,15 +247,67 @@ inherited frmModelEditor: TfrmModelEditor
...
+            Width = 145
+            Height = 21
+            TabOrder = 2
+            OnChange = cbInsertAfterWordClick
+            OnClick = cbInsertAfterWordClick
+            OnKeyUp = cbInsertAfterWordKeyUp
+            Items.Strings = (
+              '(space U+0020)'
+              '(no word break)'
+              '(Tibetan tsheg U+0F0B)'
```
Comment (darcywong00) [inline windows/src/developer/TIKE/child/Keyman.Developer.UI.UfrmModelEditor.dfm:271]: Does this string array also need zero-width space?

**Review by jahorton — APPROVED** [review]

File: windows/src/developer/TIKE/child/Keyman.Developer.UI.UfrmModelEditor.dfm (hunk 4 above)
Comment (mcdurdin) [reply windows/src/developer/TIKE/child/Keyman.Developer.UI.UfrmModelEditor.dfm:271]: I should remove those -- they are filled at runtime.

File: windows/src/developer/TIKE/child/Keyman.Developer.UI.UfrmModelEditor.dfm (hunk 3 above)
Comment (mcdurdin) [reply windows/src/developer/TIKE/child/Keyman.Developer.UI.UfrmModelEditor.dfm:222]: Good catch!

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 14.0.251-beta](https://keyman.com/downloads/releases/beta/14.0.251)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.19-alpha](https://keyman.com/downloads/releases/alpha/15.0.19)

**Pull request merged**